src/scenes/App/components/TermsAndConditions/index.js

```diff
@@ -0,0 +1,265 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+import { connect } from 'react-redux';
+import { Paper, Typography, Grid, FormGroup, FormControlLabel, Checkbox, Button, withStyles } from 'material-ui';
+import { FormattedMessage, injectIntl, intlShape } from 'react-intl';
+
+import AppConfig from '../../../../config/app';
+import styles from './styles';
+
+const TNC_ACCEPTED = 'tncAccepted';
+
+@injectIntl
+@withStyles(styles, { withTheme: true })
+export default class TermsAndConditions extends Component {
+  static propTypes = {
+    classes: PropTypes.object.isRequired,
+    intl: intlShape.isRequired, // eslint-disable-line react/no-typos
+    langHandler: PropTypes.func,
+  };
+
+  static defaultProps = {
+    langHandler: undefined,
+  };
+
+  state = {
+    isVisible: !!localStorage.getItem(TNC_ACCEPTED),
+    checkboxChecked: false,
+  };
+
+  render() {
+    const { classes, intl: { locale } } = this.props;
+    const { isVisible, checkboxChecked } = this.state;
+
+    return (
+      <div
+        className={classes.tncBg}
+        style={{
+          opacity: isVisible ? 0 : 1,
+          display: isVisible ? 'none' : 'block',
+        }}
+      >
+        <Grid container>
+          <Grid item xs>
+            <Button color="primary" variant="raised" className={classes.langBtn} onClick={this.props.langHandler}>
+              <FormattedMessage id="language.select" defaultMessage="中文" />
```

Comment:
do we need to use the `<FormattedMessage />` component here? Can't we use `intl.formatMessage`? One less thing to have to import

Reply:
oh interesting. Ok 😃